api/src/main/scala/org/virtuslab/ideprobe/EndpointsConfig.scala

```diff
@@ -0,0 +1,15 @@
+package org.virtuslab.ideprobe
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+case class EndpointsConfig(awaitIdle: EndpointsConfig.AwaitIdle)
+
+object EndpointsConfig {
+  def apply(): EndpointsConfig = EndpointsConfig(awaitIdle = AwaitIdle.Default)
+
+  case class AwaitIdle(initialWait: FiniteDuration, newTaskWait: FiniteDuration, checkFrequency: FiniteDuration)
```

Comment:
I'd put it outside of companion object and even put in a separate file as part of protocol. This is because it should also be an optional param to awaitIdle endpoint so that it can be configured per call from code.